src/dependencies/GitHubZipExtractor.ts

```diff
@@ -0,0 +1,45 @@
+import * as path from 'path';
+
+import { DependencyInstaller, Location, ProgressListener, InstallerSequence, FileDownloader, ZipExtractor } from '..';
+
+/**
+ * Extension of RemoteZipExtractor with the following features:
+ * - no remote URL needed at construction time: a (potentially expensive) computation of the remote URL is only 
+ *   performed when a download will actually take place.
+ * - the correct header for downloading a GitHub release asset is set
+ * - if a GitHub token is provided it is used to perform the download as an authenticated user
+ */
+export class GitHubZipExtractor implements DependencyInstaller {
+    private sequence: InstallerSequence | undefined;
+
+    constructor(
+        readonly remoteUrlFn: () => Promise<string>,
+        readonly folderName: string, // non optional as we need to determine whether a download is necessary
+        readonly token?: string
```

Comment:
Can it be that these are `public` by default? If it is so you could add `private`.

Reply:
You're right, `public` is the default